Scripts/Characters/Player/Player.cs

```diff
@@ -117,17 +118,40 @@ private void HandleDash()
             {
                 _animatedSprite.Stop();
                 _animatedSprite.Play("jump");
-                Position -= dashDistance;
+                StartDash(-dashDistance);
             }
             else if (_lastHorizontalInput == "right")
             {
                 _animatedSprite.Stop();
                 _animatedSprite.Play("jump");
-                Position += dashDistance;
+                StartDash(dashDistance);
             }
         }
     }
 
+    private async void StartDash(Vector2 dashDistance)
```

Comment:
Since StartDash is async, you should use `await` before calling the method (example: `await StartDash(...)`